### PR TITLE
[Impeller] Add ColorBurn, fix SourceOver alpha, make default pipeline initialization robust

### PR DESCRIPTION
**Pull request by bdero**
In addition to adding ColorBurn and activating advanced blends in the display list dispatcher, this change fixes pipeline alpha blend/default initialization problems that were resulting in double-premultiplied alpha for the SourceOver (default) Entities blend mode. This issue is difficult to notice in isolation, but rapidly compounds with overlapping transparent objects.

* Use SourceOver alpha compositing for advanced/color blends.
* Correct alpha source factor on SourceOver.
* For advanced blends, let the pipeline handle alpha compositing and set the pipeline blend mode to `SourceOver`. The advanced blend shaders just manipulate color channels and pass the source alpha onward for blending. This makes the alpha channel affect the resulting color in the correct way.
* Add ColorBurn.
* Activate `Screen` and `ColorBurn` in the display list dispatcher.
* Share vertex shader between advanced blend pipelines.
* When creating default pipelines, always decorate the descriptor with default `ContentContextOptions` options. The default pipeline blend options no longer matched `kSourceOver` (the default entities blend mode). Rather than continuing to sync the default pipeline descriptor in the render layer, it's more robust to just apply the same Entities-specific `ContentContextOptions` transform to the pipeline descriptor in all cases where a new pipeline is created.

https://user-images.githubusercontent.com/919017/168003031-df0abcc6-63fe-4b4c-9c21-370ce6daba53.mov

All blend modes now closely match this example: https://fiddle.skia.org/c/@BlendModes

Compare with the [previous video](https://github.com/flutter/engine/pull/32982#issuecomment-1115746833) I recorded of this test, where a dark outer ring can be seen around the color wheel due to the alpha compositing issues.

### CODE REVIEW

File: impeller/entity/BUILD.gn [hunk 1]
```diff
@@ -14,10 +14,11 @@ impeller_shaders("entity_shaders") {
     "shaders/solid_fill.vert",
     "shaders/solid_stroke.frag",
     "shaders/solid_stroke.vert",
-    "shaders/texture_blend.frag",
-    "shaders/texture_blend.vert",
-    "shaders/texture_blend_screen.frag",
-    "shaders/texture_blend_screen.vert",
+    "shaders/blend.frag",
```
Comment (zanderso) [inline impeller/entity/BUILD.gn:17]: nit: please alphabetize this list.

**Review by zanderso — COMMENTED** [review]
License file needs to be updated.

File: impeller/entity/BUILD.gn (hunk 1 above)
Comment (bdero) [reply impeller/entity/BUILD.gn]: Done.

**Comment by bdero** [comment]
Ended up changing the `static ContentContext::ApplyOptionsToDescriptor(pipeline, options)` to `ContentContextOptions::ApplyToPipelineDescriptor(pipeline) const` and moved definition out of the header.

**Pull request merged**